CMakeLists.txt

```diff
@@ -15,25 +15,69 @@ add_subdirectory(source)
 add_subdirectory(3rdparty/software)
 add_subdirectory(3rdparty/plutovg)
 
-if(BUILD_SHARED_LIBS)
-    target_compile_definitions(lunasvg PUBLIC LUNASVG_SHARED)
-    target_compile_definitions(lunasvg PRIVATE LUNASVG_EXPORT)
+set_target_properties(lunasvg
+    PROPERTIES
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN 1
+)
+
+add_library(lunasvg::lunasvg ALIAS lunasvg)
+
+if(NOT BUILD_SHARED_LIBS)
+    set_target_properties(lunasvg
+        PROPERTIES
+        COMPILE_FLAGS -DLUNARSVG_STATIC_DEFINE
+    )
 endif()
 
 if(LUNASVG_BUILD_EXAMPLES)
     add_subdirectory(example)
 endif()
 
-set(LUNASVG_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib)
-set(LUNASVG_INCDIR ${CMAKE_INSTALL_PREFIX}/include)
+#
+# Installation
+#
+include(GNUInstallDirs)
+install(
+    FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/lunasvg.h
+    ${CMAKE_CURRENT_BINARY_DIR}/include/lunasvg_export.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lunasvg
+)
+install(
+    TARGETS lunasvg
+    EXPORT lunasvg-targets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+install(
+    EXPORT lunasvg-targets
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/lunasvg NAMESPACE lunasvg:: FILE lunasvgTargets.cmake
+)
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/lunasvgConfig.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/lunasvgConfig.cmake
+    INSTALL_DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}/cmake/lunasvg
+)
 
-install(FILES
-    include/lunasvg.h
-    DESTINATION ${LUNASVG_INCDIR}
+write_basic_package_version_file(lunasvgConfigVersion.cmake
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
 )
 
-install(TARGETS lunasvg
-    LIBRARY     DESTINATION    ${LUNASVG_LIBDIR}
-    ARCHIVE     DESTINATION    ${LUNASVG_LIBDIR}
-    INCLUDES    DESTINATION    ${LUNASVG_INCDIR}
+install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/lunasvgConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/lunasvgConfigVersion.cmake
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/lunasvg
 )
+
+export(EXPORT lunasvg-targets FILE ${CMAKE_CURRENT_BINARY_DIR}/lunasvgTargets.cmake NAMESPACE lunasvg::)
+
+# Register package in user's package registry
+export(PACKAGE lunasvg)
```

Comment:
I think we don't usually recommend to do this in FOSS projects, I don't see it as a very popular feature in CMake. Sometimes it comes as a surprise that `find_package` actually searches the user's package registry.

Reply:
Thanks, will go with 3.13 as discussed.

Reply:
OK will remove.